File: files/CH-047.py
import pandas as pd

input = pd.read_excel("CH-047 Multiple text replaces.xlsx", sheet_name="Sheet1", usecols="B", skiprows=1)
dict = pd.read_excel("CH-047 Multiple text replaces.xlsx", sheet_name="Sheet1", usecols="E:F", skiprows=1).fillna(" ")
test = pd.read_excel("CH-047 Multiple text replaces.xlsx", sheet_name="Sheet1", usecols="J", skiprows=1)
test.columns = input.columns

for index, row in dict.iterrows():
    input["Product IDs"] = input["Product IDs"].str.replace(row[0], row[1])

print(input.equals(test)) # True

File: files/000-099/047/CH-047.R
library(tidyverse)
library(readxl)

input = read_excel("files/CH-047 Multiple text replaces.xlsx", range = "B2:B11")
dict  = read_excel("files/CH-047 Multiple text replaces.xlsx", range = "E2:F10") %>%
  replace_na(list(Old = " "))
test  = read_excel("files/CH-047 Multiple text replaces.xlsx", range = "J2:J11")

result = input$`Product IDs` %>%
  reduce(dict$Old, ~ str_replace_all(.x, fixed(.y), dict$New[dict$Old == .y]), .init = .) %>%
  tibble(`Product IDs` = .)

identical(result, test)
# [1] TRUE